utilities/transactions/write_prepared_txn_db.h

```diff
@@ -153,10 +153,21 @@ class WritePreparedTxnDB : public PessimisticTransactionDB {
                         prep_seq, snapshot_seq, 1, min_uncommitted);
       return true;
     }
+    // Commit of delayed prepared has two non-atomic steps: add to commit cache,
+    // remove from delayed prepared. Our reads from these two is also
+    // non-atomic. By looking into commit cache first thus we might not find the
+    // prep_seq neither in commit cache not in delayed_prepared_. To fix that i)
+    // we check if there was any delayed prepared BEFORE looking into commit
+    // cache, ii) if there was, we complete the search steps to be these: i)
+    // commit cache, ii) delayed prepared, commit cache again. In this way if
+    // the first query to commit cache missed the commit, the 2nd will catch it.
+    bool was_empty = delayed_prepared_empty_.load(std::memory_order_acquire);
```

Comment:
Is there a reason we had to move this check up? Is it just to be extra safe?

Reply:
Consider this order: a1) read commit_cache b1) write(commit_cache), b2) remove delayed prepared,  a2) read delayed_prepared_
In that case the reader (a) does not see either the commit in the commit cache or the prepared in delayed_prepared_ and assumes the data was committed long time ago. By checking if delayed_prepared_ is empty before looking into commit cache, this case is avoided.